### ISSUE TITLE
Artefacts with overlapping pies

### ISSUE THREAD

**Issue by dramanica** [YuLab-SMU/scatterpie#32]
In some instances, ovelapping pies are clipped incorrectly. Here is a simple reproducible example:
```
df<-data.frame(lat=c(1,1.5),long=c(2,2.1),A=c(0.2,0.5),B=c(0.7,0.4),C=c(0.1,0.1))
ggplot()+geom_scatterpie(aes(x=long,y=lat),cols=c("A","B","C"),data=df,pie_scale=150)+coord_equal()

```
![image](https://user-images.githubusercontent.com/3873428/117030999-1474d500-acf8-11eb-93d5-4868dff5a915.png)

**Comment by alexkrohn** [comment]
I've found this same issue and haven't found a solution. From what I can tell it occurs because the colors are plotted sequentially instead of by pies (e.g. all green slices are plotted, then all red slices, then all blue slices, so the blues overlap). I'm not sure how to remedy this. 

@dramanica Did you find a solution to this?

**Comment by dramanica** [comment]
@alexkrohn No, I am afraid I never did find a proper solution; swapping the order of the factor levels can help depending on how the colours overlap (but that's a hack that will only work under some circumstances).

**Comment by alexkrohn** [comment]
Bummer. I love this package, but this error is forcing me back into base R. Thanks for looking into it, anyway!

**Comment by dicorynia** [comment]
Adding an id column and using it for grouping solves this issue
```
df <- data.frame(id = c(1, 2), 
                 lat=c(1,1.5),
                 long=c(2,2.1),
                 A=c(0.2,0.5),
                 B=c(0.7,0.4),
                 C=c(0.1,0.1))

ggplot()+
  geom_scatterpie(data=df,
                  aes(x=long,y=lat, group = id),
                  cols=c("A","B","C"),
                  pie_scale=150)+
  coord_equal()
```
![image](https://user-images.githubusercontent.com/31208891/221898519-b0955737-ec87-4e95-823e-3cebca52eba7.png)

**Comment by GuangchuangYu** [comment]
@xiangpin pls incorporate this feature (using `.id` internally) to the `geom_scatterpie` layer.
